### ISSUE TITLE
WiFi hostname of 18 characters causes DHCP OPTIONS assert due to VCI default value (IDFGH-6059)

### ISSUE THREAD

**Issue by atanisoft** [espressif/esp-idf#7742]
## Environment

- Development Kit:      ESP32-S3 (any ESP32 should reproduce this though)
- Kit version (for WroverKit/PicoKit/DevKitC): v1
- Module or chip used:  ESP32-S3 WROOM 8mb / 2mb
- IDF version:  v4.4-dev-3540-g4e03a9c34c
- Build System:         idf.py
- Compiler version: xtensa-esp32s3-elf-gcc (crosstool-NG esp-2021r1) 8.4.0
- Operating System:     Linux
- Using an IDE?: No
- Power Supply:         USB

## Problem Description

After https://github.com/espressif/esp-idf/issues/6786 was merged via https://github.com/espressif/esp-lwip/commit/ae7edc2a93f7163e9858c6a6b03ff3aa077bd928 it introduced a new vendor class identifier which defaults to the hostname of the device.  This commit to esp-lwip was picked up via https://github.com/espressif/esp-idf/commit/e451a9b2e16c2607a6f84eef015ce4afc2ebdb8f.

When the hostname is 17 characters or shorter this works fine but when the hostname is 18 characters it causes an assert as seen below.

Since this optional functionality is not required by all users it would be better to make this optional via sdkconfig so those that require this functionality can enable it selectively. Additionally it would be better to have no default payload when the user does not provide one and instead skip the inclusion entirely.

### Expected Behavior

Hostname length of up to 32 characters should be supported still.

### Actual Behavior

```
[WiFi] Setting ESP32 hostname to "esp32cs_5020103fff".
[WiFi] MAC Address: 7c:df:a1:e0:59:90
[DHCP] Starting DHCP Client.
[WiFi] Station started, attempting to connect to SSID: XXXXXXX.
Allocating new alias 436 for node 05020103fffe

assert failed: dhcp_option_vendor_class_identifier IDF/components/lwip/lwip/src/core/ipv4/dhcp.c:1857 (DHCP: hostname is too long!)

Backtrace:0x40375d8a:0x3fce9e600x4037da79:0x3fce9e80 0x40385521:0x3fce9ea0 0x420a6fb2:0x3fce9fd0 0x420a723a:0x3fce9ff0 0x420a7ecf:0x3fcea020 0x420a8199:0x3fcea040 0x4209a4b1:0x3fcea060 0x4209a442:0x3fcea080 0x4209f4cd:0x3fcea0a0 0x4209f570:0x3fcea0c0 0x40381135:0x3fcea0f0 
0x40375d8a: panic_abort at esp-idf/components/esp_system/panic.c:402

0x4037da79: esp_system_abort at esp-idf/components/esp_system/esp_system.c:129
0x40385521: __assert_func at esp-idf/components/newlib/assert.c:85
0x420a6fb2: dhcp_option_vendor_class_identifier at esp-idf/components/lwip/lwip/src/core/ipv4/dhcp.c:1857 (discriminator 1)
0x420a723a: dhcp_reboot at esp-idf/components/lwip/lwip/src/core/ipv4/dhcp.c:1547
0x420a7ecf: dhcp_network_changed at esp-idf/components/lwip/lwip/src/core/ipv4/dhcp.c:1018
0x420a8199: dhcp_start at esp-idf/components/lwip/lwip/src/core/ipv4/dhcp.c:935
0x4209a4b1: esp_netif_dhcpc_start_api at esp-idf/components/esp_netif/lwip/esp_netif_lwip.c:1092
0x4209a442: esp_netif_api_cb at esp-idf/components/esp_netif/lwip/esp_netif_lwip.c:124
0x4209f4cd: tcpip_thread_handle_msg at esp-idf/components/lwip/lwip/src/api/tcpip.c:168
0x4209f570: tcpip_thread at esp-idf/components/lwip/lwip/src/api/tcpip.c:154
0x40381135: vPortTaskWrapper at esp-idf/components/freertos/port/xtensa/port.c:130
```

### Steps to reproduce

1. Use https://github.com/espressif/esp-idf/tree/master/examples/wifi/getting_started/station example configure the default hostname to be at least 18 characters or even up to 32 characters.
2. Connect to AP that supports VCI and wait for connection to complete.

**Comment by AxelLin** [comment]
> Arduino-esp32 users don't have menu config. So the effective hostname length is now 17 characters and not the documented 32.
> 
> Either the default limit for this should be updated _or_ disable VCI by default since that also is not needed/appropriate for all users.

@ESP-YJM 
I think this is a regression: enable VCI has unexpected side-effect and now VCI is enabled by default.

**Comment by atanisoft** [comment]
Thank you, it seems like a great feature to have. Perhaps move it to disabled by default and enable it via menu config and when enabled bump the options length by 32 (Max hostname length).

**Comment by AxelLin** [comment]
This should be fixed by 89873937eb7e36abc8e94071e67c9771b71d3625

**Comment by atanisoft** [comment]
@AxelLin Yes, that would appear to fix it. They did not link that commit to this issue unfortunately.
